### PR TITLE
CVM-1972 Get root hash from extended attributes when setting the publisher

### PR DESCRIPTION
**Pull request by aandvalenzuela**

### CODE REVIEW

**Comment by aandvalenzuela** [comment]
@cernvm-bot cloudtest

**Comment by aandvalenzuela** [comment]
@cernvm-bot cloudtest

**Comment by aandvalenzuela** [comment]
@cernvm-bot cloudtest

**Comment by jblomer** [comment]
Test 536 seems to fail with the patch

File: cvmfs/publish/settings.cc [hunk 1]
```diff
@@ -448,6 +448,17 @@ SettingsRepository SettingsBuilder::CreateSettingsRepository(
...
 }
 
+std::string SettingsPublisher::GetRootHashXAttr()
+{
+  std::string xattr;
+  bool rvb = platform_getxattr(
+    this->transaction().spool_area().readonly_mnt(),
+    "user.root_hash", &xattr);
+  if (!rvb) {
+    throw EPublish("cannot get extrended attribute root_hash");
```
Comment (jblomer) [inline cvmfs/publish/settings.cc]:
```suggestion
    throw EPublish("cannot get extended attribute root_hash");
```

**Comment by aandvalenzuela** [comment]
@cernvm-bot cloudtest

**Review by jblomer — CHANGES_REQUESTED** [review]
Thanks!  I think that catching the error is yet at the wrong level.

If we ask for the root hash inside the enter shell, we do expect `GetRootHashXAttr` to succeed. Only for the managed publisher, we can ignore the error.

So `GetRootHashXAttr` should still throw an exception, which is (only) caught and ignored in the lower part of `CreateSettingsPublisher`

**Comment by aandvalenzuela** [comment]
@cernvm-bot cloudtest

**Comment by aandvalenzuela** [comment]
@cernvm-bot cloudtest

**Comment by aandvalenzuela** [comment]
@cernvm-bot cloudtest cc7

**Comment by aandvalenzuela** [comment]
@cernvm-bot cloudtest cc7

**Comment by aandvalenzuela** [comment]
@cernvm-bot cloudtest cc7

**Comment by aandvalenzuela** [comment]
@cernvm-bot cloudtest cc7

**Comment by aandvalenzuela** [comment]
@cernvm-bot cloudtest

**Comment by aandvalenzuela** [comment]
@cernvm-bot cloudtest cc7

**Comment by aandvalenzuela** [comment]
@cernvm-bot cloudtest

**Comment by aandvalenzuela** [comment]
@cernvm-bot cloudtest

**Comment by aandvalenzuela** [comment]
@cernvm-bot cloudtest full

**Comment by aandvalenzuela** [comment]
@cernvm-bot cloudtest

File: test/src/683-diffworktree/main [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  fi
+
+  echo "*** Empty transaction for saving settings"
+  empty_transaction
+
+  echo "*** Transaction with rmd160 algorithm"
+  diff_command_worktree
+
+  return 0
+}
```
Comment (jblomer) [inline test/src/683-diffworktree/main]: The trailing newline is missing

**Review by jblomer — APPROVED** [review]
Looks good to me!

**Pull request merged**